metpy/plots/skewt.py

```diff
@@ -450,15 +474,14 @@ def plot_dry_adiabats(self, t0=None, p=None, **kwargs):
         """
         # Determine set of starting temps if necessary
         if t0 is None:
-            xmin, xmax = self.ax.get_xlim()
-            t0 = np.arange(xmin, xmax + 1, 10) * units.degC
+            t0 = self.get_t0()
 
         # Get pressure levels based on ylims if necessary
         if p is None:
             p = np.linspace(*self.ax.get_ylim()) * units.mbar
 
         # Assemble into data for plotting
-        t = dry_lapse(p, t0[:, np.newaxis], 1000. * units.mbar).to(units.degC)
+        t = dry_lapse(p, t0[:, np.newaxis], 1000. * units.mbar).to(t0.units)
```

Comment:
Should we be doing a similar check and conversion for pressure? While use of something other than hPa/mb is much less common than C/K, it's probably better not to assume anything? You could incorporate both checks into the method created in this PR, and rename it to `get_units` or something similar.

Reply:
I agree, I keep running into issues when I forget to change my pressure dimension from Pa to hPa. Is it OK to do a third commit?

Reply:
actually issues arise when using anything other than mbar... I will add mbar requirement to the skewt plot and subroutines if you think that's a good idea.

Reply:
Ah sorry I wasn't clear here. I'm guessing you were trying to have flexible units within the `dry_lapse` function? We would need to just handle pressure in the conditional above:

```
if p is None:
	            p = np.linspace(*self.ax.get_ylim()) * units.mbar
```
so that instead of attaching `mbar` here, we're attaching the y-axis units.

Reply:
And a third commit is fine for this change